criu/cr-restore.c

```diff
@@ -418,6 +418,10 @@ static int populate_pid_proc(void)
 		pr_err("Can't open PROC_SELF\n");
 		return -1;
 	}
+	if (open_pid_proc(PROC_SELF) < 0) {
```

Comment:
Hmmm.  (open_pid_proc(vpid(current) does the same thing, doesn't it?

Reply:
Yes I think it does. But "open_pid_proc(PROC_SELF)" is more general as it does the thing in any pidns-proc host/criu/dumpie which contains the process. AFAICR there are some places where we swap proc service fd to criu/dumpie pidns-proc so one can be confused.

Reply:
Why do we need to open vpid(current) and PROC_SELF here. Why is PROC_SELF not enough?

Reply:
Probably I'm to late here with answer, but for history:

We open both PROC_SELF_FD_OFF and PROC_PID_FD_OFF both here so that one would be able to reopen them in service fd protected context. (in sfd protected context we don't allow to open new service fd - only replace if it was already open)

Mostly we use PROC_PID_FD_OFF for opening vpid(current)'s proc, but in general it can be used to open some other pid's proc, so probably we still need it.